File: Week11/Counter.py
class Counter:
    def __init__(self, name):
        self._name = name
        self._count = 0

    def increment(self):
        self._count += 1
        return self._count
    
    def reset(self):
        self._count = 0
        return self._count
    
    @property
    def tick(self):
        return self._count
    
    @property
    def name(self):
        return self._name
    
    @name.setter
    def name(self, value):
        self._name = value

File: Week3/Task_3.1P/3.1 - Answer/CounterTask/Counter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Security.Cryptography;
using System.Text;
using System.Threading.Tasks;

namespace CounterTask
{
    public class Counter
    {
        private string _name;
        public string Name 
        {
            get => _name;
            set => _name = value;
        }

        private int _count;
        public int Tick
        {
            get => _count;
        }

        public Counter(string name)
        {
            _name = name;
            _count = 0;
        }
        public int Increment()
        {
            return _count++;
        }
        public int Reset()
        {
            return _count = 0;
        }

    }
}
